caddyhttp/httpserver/plugin.go

```diff
@@ -311,10 +313,11 @@ func (a Address) String() string {
 // VHost returns a sensible concatenation of Host:Port/Path from a.
 // It's basically the a.Original but without the scheme.
 func (a Address) VHost() string {
-	if idx := strings.Index(a.Original, "://"); idx > -1 {
-		return a.Original[idx+3:]
+	vhost := a.String()
+	if idx := strings.Index(vhost, "://"); idx > -1 {
+		return vhost[idx+3:]
 	}
-	return a.Original
+	return vhost
```

Comment:
We couldn't use `Address.String()` here, because it contains additional scheme and path if not supplied.

Why not just translate non-ascii characters in `a.Original` ?